File: Raven.Client.Lightweight/Listeners/IDocumentConflictListener.cs
using Raven.Abstractions.Data;

namespace Raven.Client.Listeners
{
	/// <summary>
	/// Hooks for users that allows you to handle document replication conflicts
	/// </summary>
	public interface IDocumentConflictListener
	{
		bool TryResolveConflict(string key, JsonDocument[] conflictedDocs, out JsonDocument resolvedDocument);
	}
}

File: Raven.Client.Java/lib/src/main/java/raven/client/listeners/IDocumentConflictListener.java
package raven.client.listeners;

import java.util.List;

import raven.abstractions.basic.Holder;
import raven.abstractions.data.JsonDocument;

/**
 * Hooks for users that allows you to handle document replication conflicts
 */
public interface IDocumentConflictListener {

  public boolean tryResolveConflict(String key, List<JsonDocument> results, Holder<JsonDocument> resolvedDocument);
}
